lib/plugin.js

```diff
@@ -382,7 +420,40 @@ async function _processPosRemove (options, doc) {
   const changes = DeepDiff.getDiff({ old }, schema, options)
   const docMapped = _mappedFieldsLog({ changes, method, module, documentNumber, user })
 
+  options.connectionUri = this.model.db._connectionString
   await _saveHistoryLog(docMapped, options)
 }
 
+
+function hasMongoDuplicateProtection(self) {
+  const mongoVersion = self.model.db._connectionOptions.driverInfo.version
+  const majorVersion = mongoVersion.split(".")[0]
+
+  if(parseInt(majorVersion) >= 6) {
+    return true
+  } else {
+    return false
+  }
+}
+
+function getObject(target, dotNotation, value) {
```

Comment:
looking at it maybe this function should be called `convertDotNotationToObjects`